src/src/Controller/RssFeedController.php

```diff
@@ -0,0 +1,152 @@
+{
+    /**
+     * @var RssFeedProductsCollection
+     */
+    private $collection;
+
+    public function __construct()
+    {
+        $this->collection = new RssFeedProductsCollection();
+    }
+
+    public function generateFeed(string $baseUrl, mixed $categoryId, int $limit, string $sortBy, string $sortOrder): string
+    {
+        $xml = new \DOMDocument('1.0', 'UTF-8');
+        $xml->formatOutput = true;
+
+        $rss = $xml->createElement('rss');
+        $rss->setAttribute('version', '2.0');
+        $rss->setAttributeNS('http://www.w3.org/2000/xmlns/', 'xmlns:smly', 'https://sendsmaily.net/schema/editor/rss.xsd');
+        $rss->setAttributeNS('http://www.w3.org/2000/xmlns/', 'xmlns:atom', 'http://www.w3.org/2005/Atom');
+        $xml->appendChild($rss);
+
+        $channel = $xml->createElement('channel');
+
+        $atomLink = $xml->createElement('atom:link');
+        $atomLink->setAttribute('href', Rss::buildRssUrl($categoryId ? $categoryId : '', strval($limit), $sortBy, $sortOrder));
+        $atomLink->setAttribute('rel', 'self');
+        $atomLink->setAttribute('type', 'application/rss+xml');
+        $channel->appendChild($atomLink);
+
+        $title = $xml->createElement('title');
+        $title->appendChild($xml->createCDATASection('Store'));
+        $channel->appendChild($title);
+
+        $link = $xml->createElement('link');
+        $link->appendChild($xml->createCDATASection(htmlspecialchars($baseUrl)));
+        $channel->appendChild($link);
+
+        $description = $xml->createElement('description');
+        $description->appendChild($xml->createCDATASection('Product Feed'));
+        $channel->appendChild($description);
+
+        $lastBuildDate = $xml->createElement('lastBuildDate');
+        $lastBuildDate->appendChild($xml->createCDATASection(gmdate(DATE_RFC2822, strtotime(date('D, d M Y H:i:s')))));
+        $channel->appendChild($lastBuildDate);
+
+        $rss->appendChild($channel);
+
+        $products = $this->collection->getProducts($categoryId, $limit, $sortBy, $sortOrder);
+        foreach ($products as $product) {
+            $channel->appendChild($this->generateItemNode($xml, $baseUrl, $product));
+        }
+
+        return $xml->saveXML();
+    }
+
+    private function generateItemNode(\DOMDocument $xml, string $baseUrl, array $product): \DOMNode
+    {
+        $prod = new \Product($product['id_product']);
+
+        $item = $xml->createElement('item');
+
+        $title = $xml->createElement('title');
+        $title->appendChild($xml->createCDATASection($product['name']));
+        $item->appendChild($title);
+
+        $link = $xml->createElement('link');
+        $link->appendChild($xml->createCDATASection($prod->getLink()));
+        $item->appendChild($link);
+
+        $gUID = $xml->createElement('guid');
+        $gUID->setAttribute('isPermaLink', 'true');
+        $gUID->appendChild($xml->createCDATASection($baseUrl));
+        $item->appendChild($gUID);
```

Comment:
```suggestion
        $gUID = $xml->createElement('guid');
        $gUID->setAttribute('isPermaLink', 'true');
        $gUID->appendChild($xml->createCDATASection($prod->getLink()));
        $item->appendChild($gUID);
```

GUID permalinks must be unique and point to the item. Otherwise the XML looks good.

Reply:
Done https://github.com/sendsmaily/smaily-prestashop-module/pull/57/commits/c68b77a1f9851d79e0c59a0cd28397aa00376904